RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitPhase2WeightsProducerGPU.cc

```diff
@@ -0,0 +1,117 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "HeterogeneousCore/CUDACore/interface/ScopedContext.h"
+
+#include "DataFormats/EcalDigi/interface/EcalDataFrame_Ph2.h"
+
+#include "EcalUncalibRecHitPhase2WeightsAlgoGPU.h"
+#include "DeclsForKernelsPh2.h"
+
+class EcalUncalibRecHitPhase2WeightsProducerGPU : public edm::stream::EDProducer<edm::ExternalWork> {
+public:
+  explicit EcalUncalibRecHitPhase2WeightsProducerGPU(edm::ParameterSet const &ps);
+  ~EcalUncalibRecHitPhase2WeightsProducerGPU() override = default;
+  static void fillDescriptions(edm::ConfigurationDescriptions &);
+
+private:
+  void acquire(edm::Event const &, edm::EventSetup const &, edm::WaitingTaskWithArenaHolder) override;
+  void produce(edm::Event &, edm::EventSetup const &) override;
+
+private:
+  const std::vector<double> weights_;
```

Comment:
Using e.g.
```suggestion
  const std::vector<double, cms::cuda::HostAllocator<double>> weights_;
```
would allow the CUDA runtime to avoid copying the data into pinned host memory for the use in `cudaMemcpyAsync()`.